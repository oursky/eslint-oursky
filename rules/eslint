accessor-pairs
array-callback-return
arrow-body-style
block-scoped-var
camelcase
capitalized-comments
class-methods-use-this
complexity
consistent-return
consistent-this
constructor-super
curly
default-case
default-case-last
default-param-last
dot-notation
eqeqeq
for-direction
func-name-matching
func-names
func-style
getter-return
grouped-accessor-pairs
guard-for-in
id-denylist
id-length
id-match
init-declarations
line-comment-position
logical-assignment-operators
max-classes-per-file
max-depth
max-lines
max-lines-per-function
max-nested-callbacks
max-params
max-statements
multiline-comment-style
new-cap
no-alert
no-array-constructor
no-async-promise-executor
no-await-in-loop
no-bitwise
no-caller
no-case-declarations
no-class-assign
no-compare-neg-zero
no-cond-assign
no-console
no-const-assign
no-constant-binary-expression
no-constant-condition
no-constructor-return
no-continue
no-control-regex
no-debugger
no-delete-var
no-div-regex
no-dupe-args
no-dupe-class-members
no-dupe-else-if
no-dupe-keys
no-duplicate-case
no-duplicate-imports
no-else-return
no-empty
no-empty-character-class
no-empty-function
no-empty-pattern
no-empty-static-block
no-eq-null
no-eval
no-ex-assign
no-extend-native
no-extra-bind
no-extra-boolean-cast
no-extra-label
no-fallthrough
no-func-assign
no-global-assign
no-implicit-coercion
no-implicit-globals
no-implied-eval
no-import-assign
no-inline-comments
no-inner-declarations
no-invalid-regexp
no-invalid-this
no-irregular-whitespace
no-iterator
no-label-var
no-labels
no-lone-blocks
no-lonely-if
no-loop-func
no-loss-of-precision
no-magic-numbers
no-misleading-character-class
no-multi-assign
no-multi-str
no-negated-condition
no-nested-ternary
no-new
no-new-func
no-new-native-nonconstructor
no-new-symbol
no-new-wrappers
no-nonoctal-decimal-escape
no-obj-calls
no-object-constructor
no-octal
no-octal-escape
no-param-reassign
no-plusplus
no-promise-executor-return
no-proto
no-prototype-builtins
no-redeclare
no-regex-spaces
no-restricted-exports
no-restricted-globals
no-restricted-imports
no-restricted-properties
no-restricted-syntax
no-return-assign
no-script-url
no-self-assign
no-self-compare
no-sequences
no-setter-return
no-shadow
no-shadow-restricted-names
no-sparse-arrays
no-template-curly-in-string
no-ternary
no-this-before-super
no-throw-literal
no-undef
no-undef-init
no-undefined
no-underscore-dangle
no-unexpected-multiline
no-unmodified-loop-condition
no-unneeded-ternary
no-unreachable
no-unreachable-loop
no-unsafe-finally
no-unsafe-negation
no-unsafe-optional-chaining
no-unused-expressions
no-unused-labels
no-unused-private-class-members
no-unused-vars
no-use-before-define
no-useless-backreference
no-useless-call
no-useless-catch
no-useless-computed-key
no-useless-concat
no-useless-constructor
no-useless-escape
no-useless-rename
no-useless-return
no-var
no-void
no-warning-comments
no-with
object-shorthand
one-var
operator-assignment
prefer-arrow-callback
prefer-const
prefer-destructuring
prefer-exponentiation-operator
prefer-named-capture-group
prefer-numeric-literals
prefer-object-has-own
prefer-object-spread
prefer-promise-reject-errors
prefer-regex-literals
prefer-rest-params
prefer-spread
prefer-template
radix
require-atomic-updates
require-await
require-unicode-regexp
require-yield
sort-imports
sort-keys
sort-vars
strict
symbol-description
unicode-bom
use-isnan
valid-typeof
vars-on-top
yoda
