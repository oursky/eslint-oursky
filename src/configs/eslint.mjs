export default {
  rules: {
    "accessor-pairs": "error",
    "array-callback-return": "error",
    "arrow-body-style": "off",
    "block-scoped-var": "error",
    camelcase: "off",
    "capitalized-comments": "off",
    "class-methods-use-this": "error",
    complexity: ["error", { max: 10 }],
    "consistent-this": "off",
    "consistent-return": "error",
    "constructor-super": "error",
    curly: "off",
    "default-case": "error",
    "default-case-last": "error",
    "default-param-last": "error",
    "dot-notation": "off",
    eqeqeq: [
      "error",
      "always",
      {
        null: "ignore",
      },
    ],
    "for-direction": "error",
    "func-name-matching": "error",
    "func-names": "off",
    "func-style": "off",
    "getter-return": "error",
    "grouped-accessor-pairs": "off",
    "guard-for-in": "error",
    "handle-callback-err": "error",
    "id-denylist": "off",
    "id-length": "off",
    "id-match": "off",
    "init-declarations": "off",
    "line-comment-position": "off",
    "logical-assignment-operators": "off",
    "max-classes-per-file": "off",
    "max-depth": "off",
    "max-lines": "off",
    "max-lines-per-function": "off",
    "max-nested-callbacks": "off",
    "max-params": "off",
    "max-statements": "off",
    "multiline-comment-style": "off",
    "new-cap": "off",
    "no-alert": "off",
    "no-array-constructor": "error",
    "no-async-promise-executor": "error",
    "no-await-in-loop": "off",
    "no-bitwise": "off",
    "no-buffer-constructor": "error",
    "no-caller": "error",
    "no-case-declarations": "error",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-console": "off",
    "no-const-assign": "error",
    "no-constant-binary-expression": "error",
    "no-constant-condition": [
      "error",
      {
        checkLoops: false,
      },
    ],
    "no-constructor-return": "error",
    "no-continue": "off",
    "no-control-regex": "error",
    "no-debugger": "off",
    "no-delete-var": "off",
    "no-div-regex": "off",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-empty": "off",
    "no-empty-character-class": "error",
    "no-empty-function": "off",
    "no-empty-pattern": "off",
    "no-empty-static-block": "off",
    "no-eq-null": "off",
    "no-ex-assign": "error",
    "no-extra-boolean-cast": "off",
    "no-fallthrough": [
      "error",
      {
        commentPattern: "fallthrough",
      },
    ],
    "no-func-assign": "error",
    "no-global-assign": "error",
    "no-implicit-coercion": ["error", { allow: ["!!"] }],
    "no-implicit-globals": "error",
    "no-inline-comments": "off",
    "no-inner-declarations": "off",
    "no-invalid-regexp": "error",
    "no-invalid-this": "error",
    "no-irregular-whitespace": "off",
    "no-labels": "off",
    "no-lonely-if": "off",
    "no-magic-numbers": "off",
    "no-misleading-character-class": "error",
    "no-multi-assign": "off",
    "no-negated-condition": "off",
    "no-nested-ternary": "off",
    "no-new-native-nonconstructor": "error",
    "no-new-symbol": "off",
    "no-nonoctal-decimal-escape": "error",
    "no-obj-calls": "error",
    "no-object-constructor": "error",
    "no-octal": "error",
    "no-octal-escape": "error",
    "no-param-reassign": "off",
    "no-plusplus": "off",
    "no-promise-executor-return": "error",
    "no-prototype-builtins": "error",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-restricted-exports": "off",
    "no-restricted-globals": "off",
    "no-restricted-imports": "off",
    "no-restricted-properties": "off",
    "no-restricted-syntax": "off",
    "no-self-assign": "off",
    "no-shadow": "off",
    "no-shadow-restricted-names": "error",
    "no-sparse-arrays": "error",
    "no-ternary": "off",
    "no-this-before-super": "error",
    "no-undef": "error",
    "no-undefined": "off",
    "no-underscore-dangle": "off",
    "no-unexpected-multiline": "off",
    "no-unneeded-ternary": "off",
    "no-unreachable": "error",
    "no-unreachable-loop": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unsafe-optional-chaining": "error",
    "no-unused-labels": "error",
    "no-unused-vars": "error",
    "no-unused-private-class-members": "error",
    "no-use-before-define": ["error", { functions: false, classes: false }],
    "no-useless-backreference": "error",
    "no-useless-catch": "off",
    "no-useless-computed-key": "off",
    "no-useless-constructor": "error",
    "no-useless-escape": "error",
    "no-useless-rename": "error",
    "no-warning-comments": "off",
    "no-with": "error",
    "object-shorthand": "off",
    "one-var": "off",
    "operator-assignment": "off",
    "prefer-arrow-callback": "off",
    "prefer-destructuring": "off",
    "prefer-exponentiation-operator": "off",
    "prefer-named-capture-group": "off",
    "prefer-numeric-literals": "error",
    "prefer-object-spread": "error",
    "prefer-object-has-own": "off",
    "prefer-template": "off",
    "require-await": "off",
    "require-unicode-regexp": "off",
    "require-yield": "error",
    "sort-imports": "off",
    "sort-keys": "off",
    "sort-vars": "off",
    strict: "off",
    "symbol-description": "off",
    "unicode-bom": "off",
    "use-isnan": "error",
    "valid-typeof": "error",
    "vars-on-top": "off",
    yoda: "off",
    "no-dupe-else-if": "error",
    "no-duplicate-imports": [
      "error",
      {
        includeExports: true,
      },
    ],
    "no-else-return": "error",
    "no-eval": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-label": "error",
    "no-implied-eval": "error",
    "no-import-assign": "error",
    "no-iterator": "error",
    "no-label-var": "error",
    "no-lone-blocks": "error",
    "no-loop-func": "error",
    "no-loss-of-precision": "error",
    "no-mixed-requires": "error",
    "no-multi-str": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-require": "error",
    "no-new-wrappers": "error",
    "no-path-concat": "error",
    "no-proto": "error",
    "no-return-assign": "error",
    "no-return-await": "off",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-setter-return": "error",
    "no-template-curly-in-string": "error",
    "no-throw-literal": "error",
    "no-undef-init": "error",
    "no-unmodified-loop-condition": "error",
    "no-unused-expressions": "error",
    "no-useless-call": "error",
    "no-useless-concat": "error",
    "no-useless-return": "error",
    "no-var": "error",
    "no-void": "error",
    "prefer-const": "error",
    "prefer-promise-reject-errors": "error",
    "prefer-regex-literals": "error",
    "prefer-rest-params": "error",
    "prefer-spread": "error",
    radix: "error",
    "require-atomic-updates": "error",
  },
};
