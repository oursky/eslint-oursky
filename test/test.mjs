import { join, relative } from "path";
import { cwd } from "node:process";
import { ESLint } from "eslint";

function runTest(filename) {
  test(filename, async () => {
    const cli = new ESLint();

    const lintResults = await cli.lintFiles([join("src", filename)]);
    // Patch the path so that they are the same on any machine.
    for (const r of lintResults) {
      const absolute = r.filePath;
      r.filePath = relative(cwd(), absolute);
    }

    const formatter = await cli.loadFormatter("unix");
    const actual = await formatter.format(lintResults);
    expect(actual).toMatchSnapshot();
  });
}

runTest("simple.ts");
runTest("react.tsx");
