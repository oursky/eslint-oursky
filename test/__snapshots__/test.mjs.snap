// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react.tsx 1`] = `
"src/react.tsx:19:8: Missing return type on function. [Error/@typescript-eslint/explicit-module-boundary-types]
src/react.tsx:21:26: Spreading the same expression multiple times is forbidden [Error/react/jsx-props-no-spread-multi]
src/react.tsx:24:38: forwardRef is used with this component but no ref parameter is set [Error/react/forward-ref-uses-ref]
src/react.tsx:24:39: 'props' is defined but never used. Allowed unused args must match /^_/u. [Error/@typescript-eslint/no-unused-vars]

4 problems"
`;

exports[`simple.ts 1`] = `
"src/simple.ts:1:1: Unexpected var, use let or const instead. [Error/no-var]
src/simple.ts:1:8: Expected a \`const\` assertion instead of a literal type annotation. [Error/@typescript-eslint/prefer-as-const]
src/simple.ts:5:8: Missing return type on function. [Error/@typescript-eslint/explicit-module-boundary-types]
src/simple.ts:6:11: Switch is not exhaustive. Cases not matched: "b" [Error/@typescript-eslint/switch-exhaustiveness-check]
src/simple.ts:12:1: Unexpected var, use let or const instead. [Error/no-var]
src/simple.ts:12:5: 'b' is assigned a value but never used. Allowed unused vars must match /^_/u. [Error/@typescript-eslint/no-unused-vars]
src/simple.ts:12:11: Unexpected number value in conditional. An explicit zero/NaN check is required. [Error/@typescript-eslint/strict-boolean-expressions]
src/simple.ts:14:8: Function 'thisFunctionIsHardToComprehend' has a complexity of 14. Maximum allowed is 10. [Error/complexity]
src/simple.ts:14:17: Refactor this function to reduce its Cognitive Complexity from 19 to the 10 allowed. [Error/sonarjs/cognitive-complexity]
src/simple.ts:55:25: 'unused' is defined but never used. Allowed unused args must match /^_/u. [Error/@typescript-eslint/no-unused-vars]
src/simple.ts:60:15: The \`{}\` ("empty object") type allows any non-nullish value, including literals like \`0\` and \`""\`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want \`object\` instead.
- If you want a type meaning "any value", you probably want \`unknown\` instead. [Error/@typescript-eslint/no-empty-object-type]
src/simple.ts:66:15: Property bar should be declared as a class property. [Error/@typescript-eslint/parameter-properties]
src/simple.ts:67:5: This assignment is unnecessary since it is already assigned by a parameter property. [Error/@typescript-eslint/no-unnecessary-parameter-property-assignment]
src/simple.ts:71:45: Type parameter A is used only once in the function signature. [Error/@typescript-eslint/no-unnecessary-type-parameters]
src/simple.ts:75:36: The \`Function\` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type. [Error/@typescript-eslint/no-unsafe-function-type]
src/simple.ts:78:4: tsdoc-undefined-tag: The TSDoc tag "@foobar" is not defined in this configuration [Error/tsdoc/syntax]
src/simple.ts:80:35: Prefer using the primitive \`number\` as a type name, rather than the upper-cased \`Number\`. [Error/@typescript-eslint/no-wrapper-object-types]
src/simple.ts:83:4: tsdoc-missing-deprecation-message: The @deprecated block must include a deprecation message, e.g. describing the recommended alternative [Error/tsdoc/syntax]
src/simple.ts:88:3: \`deprecated\` is deprecated. [Error/@typescript-eslint/no-deprecated]
src/simple.ts:96:12: Don't use \`BrokenType\` as a type. BrokenType is broken. Use NonBrokenType instead [Error/@typescript-eslint/no-restricted-types]

20 problems"
`;
